app/com/arpnetworking/metrics/portal/reports/impl/chrome/DevToolsNetworkConfigurationProtocol.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.reports.impl.chrome;
+
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.github.kklisura.cdt.protocol.commands.Fetch;
+import com.github.kklisura.cdt.protocol.commands.Network;
+import com.github.kklisura.cdt.protocol.types.fetch.HeaderEntry;
+import com.github.kklisura.cdt.protocol.types.network.ErrorReason;
+import com.github.kklisura.cdt.protocol.types.network.Request;
+import com.github.kklisura.cdt.protocol.types.network.RequestPattern;
+import com.github.kklisura.cdt.services.ChromeDevToolsService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/* package private */ enum DevToolsNetworkConfigurationProtocol {
+
+    DEPRECATED_NETWORK(DevToolsNetworkConfigurationProtocol::configureWithNetwork),
+    FETCH(DevToolsNetworkConfigurationProtocol::configureWithFetch);
+
+    DevToolsNetworkConfigurationProtocol(final BiConsumer<ChromeDevToolsService, PerOriginConfigs> configure) {
+        _configure = configure;
+    }
+
+    public void configure(final ChromeDevToolsService dts, final PerOriginConfigs originConfigs) {
+        _configure.accept(dts, originConfigs);
+    }
+
+    private final BiConsumer<ChromeDevToolsService, PerOriginConfigs> _configure;
+
+    @SuppressWarnings("deprecation")
+    private static void configureWithNetwork(final ChromeDevToolsService dts, final PerOriginConfigs originConfigs) {
```

Comment:
(this comes from https://github.com/ArpNetworking/metrics-portal/blob/8c0fe5cd25ac3f28d1b337af5ce2155893f70b95/app/com/arpnetworking/metrics/portal/reports/impl/chrome/DevToolsServiceWrapper.java#L74 )